solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1295,16 +1307,28 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         uint256 _reserveAmount2,
         uint256 _value
     ) internal {
-        IERC20Token[] memory reserveTokens = new IERC20Token[](2);
-        uint256[] memory amounts = new uint256[](2);
-        reserveTokens[0] = _reserveToken1;
-        reserveTokens[1] = _reserveToken2;
-        amounts[0] = _reserveAmount1;
-        amounts[1] = _reserveAmount2;
-
         // ensure that the contract can receive ETH
         updatingLiquidity = true;
-        _converter.addLiquidity{ value: _value }(reserveTokens, amounts, 1);
+
+        if (isStandardPool(_converter)) {
+            IERC20Token[2] memory reserveTokens;
+            uint256[2] memory amounts;
+            reserveTokens[0] = _reserveToken1;
+            reserveTokens[1] = _reserveToken2;
+            amounts[0] = _reserveAmount1;
+            amounts[1] = _reserveAmount2;
+            _converter.addLiquidity{ value: _value }(reserveTokens, amounts, 1);
+        } else {
+            IERC20Token[] memory reserveTokens = new IERC20Token[](2);
+            uint256[] memory amounts = new uint256[](2);
+            reserveTokens[0] = _reserveToken1;
+            reserveTokens[1] = _reserveToken2;
+            amounts[0] = _reserveAmount1;
+            amounts[1] = _reserveAmount2;
+            _converter.addLiquidity{ value: _value }(reserveTokens, amounts, 1);
```

Comment:
This block looks like a code duplication

Reply:
It is not, because `IERC20Token[2]` and `IERC20Token[]` are two different types with no common "ancestor".
Same goes for `uint256[2]` and `uint256[]`.
If the language supported templates, then we could unify these two blocks.
Otherwise, I don't see how this can be done.